File: JinJiWeb/ClientSite/src/app/shared-models/order.ts
import { OrderDetail } from './order-detail';

export class Order {
    id: number;
    year: number;
    month: number;
    day: number;
    sum: number = 0;

    constructor(date: Date = null) {
        if(!date) {
            date = new Date();
        }

        this.year = date.getFullYear();
        this.month = date.getMonth() + 1;
        this.day = date.getDate();
    }

    orderDetails: OrderDetail[] = [];

    get totalPrice() {
        return this.orderDetails.map(p => p.price).reduce((p, r) => p + r, 0);
    }
}

File: JinJiWeb/Models/Order.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;

namespace JinJiWeb.Models {
    public class Order {
        [Required]
        public int Id { get; set; }

        [Required]
        public int Year { get; set; }

        [Required]
        public int Month { get; set; }

        [Required]
        public int Day { get; set; }

        public double Sum { get; set; }

        public virtual ICollection<OrderDetail> OrderDetails { get; set; }
    }
}